### ISSUE TITLE
error in the date field when editing project with locale=`it`

### ISSUE THREAD

**Issue by JohnRDOrazio** [opensourcecatholic/marriage-booklet#54]
I just noticed on the production server, that when accessing the **Edit This Project** with locale = **Italian** (i.e., [https://marriagebooklet.bibleget.io/it/projects/1/edit](https://marriagebooklet.bibleget.io/it/projects/1/edit)), an error is generated.

Inspecting the logs, I see this error message:

```bash
I, [2021-07-08T15:07:48.444654 #1625149]  INFO -- : [6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]   Rendered layout layouts/application.html.erb (Duration: 2.1ms | Allocations: 996)
I, [2021-07-08T15:07:48.444843 #1625149]  INFO -- : [6db9a2b2-66f3-4c86-a1c1-c7f4b130e779] Completed 500 Internal Server Error in 4ms (ActiveRecord: 0.1ms | Allocations: 1341)
F, [2021-07-08T15:07:48.446345 #1625149] FATAL -- : [6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779] ActionView::Template::Error (.date.order only accepts :year, :month and :day):
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     90:               <% if I18n.locale == :en then %>
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     91:               <%= form.datetime_select :weddingdate, { ampm: true }, { class: "mt-1 focus:ring-indigo-500 focus:border-indigo-500 shadow-sm sm:text-sm border-gray-300 rounded-md" } %>
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     92:               <% else %>
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     93:               <%= form.datetime_select :weddingdate, { ampm: false }, { class: "mt-1 focus:ring-indigo-500 focus:border-indigo-500 shadow-sm sm:text-sm border-gray-300 rounded-md" } %>
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     94:               <% end %>
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     95:             </div>
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]     96:           </div><!--Marriage date block end -->
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779]
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779] app/views/projects/_form.html.erb:93
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779] app/views/projects/_form.html.erb:1
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779] app/views/projects/edit.html.erb:3
[6db9a2b2-66f3-4c86-a1c1-c7f4b130e779] app/controllers/application_controller.rb:11:in `switch_locale'
```

Is it `ampm: false` that is causing this?

**Comment by JohnRDOrazio** [comment]
This is also happening on the development instance.
I see someone else getting bitten by this error, [suggests this solution](https://stackoverflow.com/a/44576961/394921).

I did set something of the sorts in the translation files, they probably need to be double checked.

**Comment by JohnRDOrazio** [comment]
Yes I see what happened. Upon inspecting the translation .yml files, I see that I had defined the order of "month, day, year" for each locale, using an array:

`en` => ["month","day","year"]

`it` => ["day", "month", "year"]

However, when using the weblate translation tool, these showed as strings that needed translation, and so wound up becoming:

`it` => ["mese","giorno","anno"]

As I recall, I needed to define this array to order them correctly, because it wasn't being handled by the I18n API out of the box.

So the trick here is going to be, use the weblate translation tool to translate them to the english correspondents in the newly desired order:

* `month` => `day`
* `day` => `month`
* `year` => `year`

The translation tool will probably complain about `year` not having been translated, but hey that's life. As long as we get the job done.
